### PR TITLE
Consume one subscription each time

### PR DESCRIPTION
**Pull request by huiwangredhat**
Fixes OLCS-531

### CODE REVIEW

File: openlcsd/flow/tasks.py [hunk 1]
```diff
@@ -1605,23 +1606,19 @@ def trigger_corgi_components_imports(context, engine):
...
     trigger_corgi_components_imports
 ]
 
 
-flow_get_corgi_components = [
+flow_get_active_subscriptions = [
     get_config,
-    get_active_subscriptions,
-    populate_components_generator,
-    WHILE(
```
Comment (rhyw) [inline openlcsd/flow/tasks.py:1621]: I'm worried about removing this - when testing, please make sure the number of components collected are as expected.

The original code attemptps to populate/consume the generator in the `WHILE` statement till it's exhausted. I don't see in this patch how it was dealt with. Assuming there could be missings, i.e., only the first 100 rpm components are processed.

File: openlcsd/flow/tasks.py [hunk 2]
```diff
@@ -1486,18 +1488,18 @@ def populate_source_components(context, engine):
...
+        source_components = next(components_generator)
     except StopIteration:
         # Stop the flow since all chunks are consumed.
         engine.stop()
     else:
         context["source_components"] = source_components
 
 
-def collect_components(subscriptions):
+def collect_components(subscription):
```
Comment (rhyw) [inline openlcsd/flow/tasks.py:1502]: Always remember to review/update the docstring if param/intention changes.

File: openlcs/libs/corgi.py [hunk 3]
```diff
@@ -612,7 +612,7 @@ def get_component_instance_from_nvr(self, nvr):
         raise ValueError(err_msg)
 
     def collect_components_from_subscriptions(self,
-                                              subscriptions,
+                                              subscription,
```
Comment (rhyw) [inline openlcs/libs/corgi.py:615]: Similar here - original params is plural. if it's updated to collect one at a time, do you think func name should be changed as well?

**Review by rhyw — COMMENTED** [review]
Please give a verification on whether there's missings in case a subscription contains hundreds of components. Also might worth give a test after deployed on OCP - using subscription in OLCS-463

File: openlcsd/flow/tasks.py (hunk 1 above)
Comment (huiwangredhat) [reply openlcsd/flow/tasks.py:1621]: Updated.

File: openlcs/libs/corgi.py (hunk 3 above)
Comment (huiwangredhat) [reply openlcs/libs/corgi.py]: Updated.

**Comment by rhyw** [comment]
Generally LGTM

**Pull request merged**